kong/plugins/key-auth/handler.lua

```diff
@@ -149,6 +150,11 @@ end
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- check if preflight request and wether it should be authenticated
+  if conf.authenticate_preflight and get_method() == "OPTIONS" then
```

Comment:
Acknowledged it was a bug, but the confusion still exists as described... I think we should use a better name for this value.